src/Shared/AppModels/AppDebug.cs

```diff
@@ -0,0 +1,229 @@
+using Windows.UI.Core;
+using Windows.UI.Popups;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace StrixMusic.AppModels
+{
+    /// <summary>
+    /// Responsible for handling debug information.
+    /// </summary>
+    public partial class AppDebug : ObservableObject
+    {
+        private SemaphoreSlim _semaphoreSlim = new SemaphoreSlim(1, 1);
+
+        [ObservableProperty]
+        private string? _windowSizeStr;
+
+        [ObservableProperty]
+        private string? _memoryUsage;
+
+        [ObservableProperty]
+        private string? _windowHeight;
+
+        [ObservableProperty]
+        private string? _windowWidth;
+
+        [ObservableProperty]
+        private string? _totalMemory;
+
+        [ObservableProperty]
+        private string? _expectedMemoryLimit;
+
+        private DispatcherTimer _memoryWatchTimer;
+        private ContentDialog? _deleteUserDataDialog;
+
+        /// <summary>
+        /// Holds the list of all logs.
+        /// </summary>
+        public ObservableCollection<string>? AppLogs { get; } = new ObservableCollection<string>();
+
+        /// <summary>
+        /// Creates a new instance of <see cref="AppDebug"/>.
+        /// </summary>
+        public AppDebug()
+        {
+            Logger.MessageReceived += Logger_MessageReceived;
+            Window.Current.SizeChanged += Current_SizeChanged;
+            MemoryManager.AppMemoryUsageIncreased += MemoryManager_MemoryChanged;
+            MemoryManager.AppMemoryUsageDecreased += MemoryManager_MemoryChanged;
+            UpdateMemoryUsage();
+            SetWindowSize();
+            _memoryWatchTimer = new DispatcherTimer();
+            _memoryWatchTimer.Interval = TimeSpan.FromSeconds(3);
+            _memoryWatchTimer.Tick += DispatchTimer_Elapsed;
+            _memoryWatchTimer.Start();
+        }
+
+        private void DispatchTimer_Elapsed(object sender, object e)
+        {
+            _memoryWatchTimer.Stop();
+            UpdateMemoryUsage();
+            _memoryWatchTimer.Start();
+        }
+
+        private void MemoryManager_MemoryChanged(object sender, object e)
+        {
+            UpdateMemoryUsage();
+        }
+
+        private void UpdateMemoryUsage()
+        {
+            var memoryUsed = MemoryManager.AppMemoryUsage;
+            var totalMemory = MemoryManager.AppMemoryUsageLimit;
+            MemoryUsage = SizeSuffix((long)memoryUsed);
+            TotalMemory = SizeSuffix((long)totalMemory);
+        }
+
+        private void Current_SizeChanged(object sender, Windows.UI.Core.WindowSizeChangedEventArgs e) => SetWindowSize();
+
+        private void SetWindowSize()
+        {
+            WindowHeight = Window.Current.Bounds.Height.ToString(); ;
+            WindowWidth = Window.Current.Bounds.Width.ToString();
+        }
+
+        private async void Logger_MessageReceived(object sender, LoggerMessageEventArgs e)
+        {
+            await _semaphoreSlim.WaitAsync();
```

Comment:
Let's replace this with the `.DisposableWaitAsync()` extension method in OwlCore.Extensions. You forgot to release the semaphore before returning a few lines down.

Doing `using (await _semaphore.DisposableWaitAsync() { ... }` will remove the need to explicitly release the semaphore.